sites/docs/content/components/element-field.md

```diff
@@ -0,0 +1,140 @@
+
+`ElementField`s should be used within the context of a [Field](/docs/components/field) or [Fieldset](/docs/components/fieldset) component and creates its own context to scope the errors and other state of the field.
+
+## Usage
+
+Here's an example of how you might use the `ElementField` component to create a dynamic list of URLs in a form.
+
+```svelte
+<script lang="ts">
+	import {
+		ElementField,
+		FieldErrors,
+		Control,
+		Label,
+		Fieldset,
+		Description,
+	} from "formsnap";
+	import { superForm } from "sveltekit-superforms";
+	import { zodClient } from "sveltekit-superforms/adapters";
+	import { z } from "zod";
+
+	const schema = z.object({
+		urls: z.array(z.string().url()).min(2).default(["", ""]),
+	});
+
+	export let data;
+
+	const form = superForm(data.form, {
+		validators: zodClient(schema),
+	});
+
+	const { form: formData, enhance } = form;
+</script>
+
+<form use:enhance method="POST">
+	<Fieldset {form} name="urls">
+		<Legend>Enter your URLS</Legend>
+		{#each $formData.urls as _, i}
+			<ElementField {form} name="urls[{i}]">
+				<Control let:attrs>
+					<Input type="url" bind:value={$formData.urls[i]} {...attrs} />
+				</Control>
+				<FieldErrors />
+			</ElementField>
+		{/each}
+		<Description>Your URLs will be displayed on your public profile.</Description>
+		<FieldErrors />
+	</Fieldset>
+	<button type="submit">Submit</button>
+</form>
+```
+
+As you can see, we're able to display errors for each index within the array, as well as array-level errors for the entire fieldset.
+
+Check out the [Dynamic Fields](/docs/recipes/dynamic-fields) recipe for more advanced usage of the `ElementField` component.
+
+## Props
+
+The `ElementField` component doesn't render an element, it strictly provides context for its children. It accepts the following props:
+
+```ts
+export type ElementFieldProps<
+	T extends Record<string, unknown>,
+	U extends FormPathLeaves<T>,
+> = {
+	/**
+	 * The form object returned from calling `superForm` in your component.
+	 */
+	form: SuperForm<T>;
+
+	/**
+	 * The path to the field in the form object.
+	 */
+	name: U;
+};
+```
+
+## Slot Props
+
+The following slot props are provided for convenience and ease of composition when using the `ElementField` component.
+
+```ts
+type SlotProps<T extends Record<string, unknown>, U extends FormPath<T>> = {
+	/** The value of the value store of the field. */
+	value: T[U];
+
+	/** The value of the errors store for the field. */
+	errors: string[] | undefined;
+
+	/** The constraints for the field. */
+	constraints: Record<string, unknown>;
+
+	/** Whether the field is tainted or not. */
+	tainted: boolean;
+};
+```
+
+## Composition
+
+Since the `ElementField` component doesn't render any HTML elements, it's a common practice to create a wrapper component around it to have consistent styling and behavior across your forms.
```

Comment:
```suggestion
Since the `ElementField` component doesn't render any HTML elements, it's common practice to create a wrapper component around it to have consistent styling and behavior across your forms.
```